src/common/utils/registry/repository.go

```diff
@@ -69,49 +69,61 @@ func parseError(err error) error {
 // ListTag ...
 func (r *Repository) ListTag() ([]string, error) {
 	tags := []string{}
-	req, err := http.NewRequest("GET", buildTagListURL(r.Endpoint.String(), r.Name), nil)
-	if err != nil {
-		return tags, err
-	}
-
-	resp, err := r.client.Do(req)
-	if err != nil {
-		return tags, parseError(err)
-	}
-
-	defer resp.Body.Close()
-
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return tags, err
-	}
-
-	if resp.StatusCode == http.StatusOK {
-		tagsResp := struct {
-			Tags []string `json:"tags"`
-		}{}
+	aurl := buildTagListURL(r.Endpoint.String(), r.Name)
 
-		if err := json.Unmarshal(b, &tagsResp); err != nil {
+	for len(aurl) > 0 {
+		req, err := http.NewRequest("GET", aurl, nil)
+		if err != nil {
 			return tags, err
 		}
-		sort.Strings(tags)
-		tags = tagsResp.Tags
-
-		return tags, nil
-	} else if resp.StatusCode == http.StatusNotFound {
+		resp, err := r.client.Do(req)
+		if err != nil {
+			return nil, parseError(err)
+		}
 
-		// TODO remove the logic if the bug of registry is fixed
-		// It's a workaround for a bug of registry: when listing tags of
-		// a repository which is being pushed, a "NAME_UNKNOWN" error will
-		// been returned, while the catalog API can list this repository.
-		return tags, nil
-	}
+		defer resp.Body.Close()
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return tags, err
+		}
 
-	return tags, &commonhttp.Error{
-		Code:    resp.StatusCode,
-		Message: string(b),
+		if resp.StatusCode == http.StatusOK {
+			tagsResp := struct {
+				Tags []string `json:"tags"`
+			}{}
+
+			if err := json.Unmarshal(b, &tagsResp); err != nil {
+				return tags, err
+			}
+
+			tags = append(tags, tagsResp.Tags...)
+			// Link: </v2/_catalog?last=library%2Fhello-world-25&n=100>; rel="next"
```

Comment:
You mean `/v2/<name>/tags/list`?

Reply:
Please update the comment here

Reply:
@bitsf Could you comment on this??

Reply:
yes, my typo, this comment should be
// Link: </v2/library/hello-world/tags/list?n=100>; rel="next"
// Link: <http://domain.com/v2/library/hello-world/tags/list?n=100>; rel="next"

Reply:
actually the url is as below
'<https://444571669415.dkr.ecr.ap-northeast-1.amazonaws.com/v2/test/nginx-photon/tags/list?last=ukD72mdD%2FmC8b5xV3susmJzzaTgp3hKwR9nRUW1yZZ4R3Jc13zaALoZQP9ut3IYattOaKlXMRmYYgRHibI9VTKT2qZ%2BXD7PV4LjLE6OjHmQkRwXypd7HXj%2FSnA9iMm3YvNsd4LmPQrO4fpYZgnZZ8rzIIYqex6%2B3A3%2FmKcTsNKkkKwRror1sjGrDqkw6MRYRWLUC5ZAd%2B7VjpfVsS5HLLEZltW%2Fl3aYbulKYdpdl0Z46tIb6CcBgfvi1ECZhEkEGPnTgU%2B5oeI8mpMftGK6i3j6U6Vn0rYXucTE2coh%2BlSCEpI6Nkzh%2F1RptJZioNqEf24qDS8t0eYzqJ%2B1%2BtWGyBvnC1qYWEFgixqfhcN8hb1g%3D>; rel="next"'

Reply:
the two comment lines should both be changed. use the simple example is enough.

// Link: </v2/library/hello-world/tags/list?last=......>; rel="next"
// Link: \<http://domain.com/v2/library/hello-world/tags/list?last=......>; rel="next"

Reply:
ok, done :)